src/Roslyn.Suppressions/Roslyn.Suppressions/SettingsFile/RoslynSettingsFileStorage.cs

```diff
@@ -19,82 +19,83 @@
+
+        /// <summary>
+        /// Return the settings for the specific settings key, or null
+        /// if there are no settings for that key
+        /// </summary>
+        RoslynSettings Get(string settingsKey);
+    }
+
+    [Export(typeof(IRoslynSettingsFileStorage))]
+    internal class RoslynSettingsFileStorage : IRoslynSettingsFileStorage
     {
         private readonly ILogger logger;
         private readonly IFileSystem fileSystem;
 
         [ImportingConstructor]
-        public SuppressedIssuesFileStorage(ILogger logger) : this(logger, new FileSystem())
+        public RoslynSettingsFileStorage(ILogger logger) : this(logger, new FileSystem())
         {
         }
 
-        internal SuppressedIssuesFileStorage(ILogger logger, IFileSystem fileSystem)
+        internal RoslynSettingsFileStorage(ILogger logger, IFileSystem fileSystem)
         {
             this.fileSystem = fileSystem;
             this.logger = logger;
             fileSystem.Directory.CreateDirectory(RoslynSettingsFileInfo.Directory);
         }
 
-        public IEnumerable<SonarQubeIssue> Get(string sonarProjectKey)
+        public RoslynSettings Get(string settingsKey)
         {
-            ValidateSonarProjectKey(sonarProjectKey);
+            Debug.Assert(settingsKey != null, "Not expecting settings to be null");
+         
             try
             {
-                var filePath = RoslynSettingsFileInfo.GetSettingsFilePath(sonarProjectKey);
+                var filePath = RoslynSettingsFileInfo.GetSettingsFilePath(settingsKey);
 
                 if(!fileSystem.File.Exists(filePath))
                 {
-                    logger.WriteLine(string.Format(Strings.SuppressedIssuesFileStorageGetError, sonarProjectKey, Strings.SuppressedIssuesFileStorageFileNotFound));
-                    return Enumerable.Empty<SonarQubeIssue>();
+                    logger.WriteLine(string.Format(Strings.SuppressedIssuesFileStorageGetError, settingsKey, Strings.SuppressedIssuesFileStorageFileNotFound));
+                    return null;
                 }
 
                 var fileContent = fileSystem.File.ReadAllText(filePath);
-                return JsonConvert.DeserializeObject<IEnumerable<SonarQubeIssue>>(fileContent);
+                return JsonConvert.DeserializeObject<RoslynSettings>(fileContent);
             }
             catch (Exception ex)
             {
-                logger.WriteLine(string.Format(Strings.SuppressedIssuesFileStorageGetError, sonarProjectKey, ex.Message));
+                logger.WriteLine(string.Format(Strings.SuppressedIssuesFileStorageGetError, settingsKey, ex.Message));
             }
 
-            return Enumerable.Empty<SonarQubeIssue>();
+            return null;
         }
 
-        public void Update(string sonarProjectKey, IEnumerable<SonarQubeIssue> allSuppressedIssues)
+        public void Update(RoslynSettings settings)
         {
-            ValidateSonarProjectKey(sonarProjectKey);
```

Comment:
The `ValidateSonarProjectKey` method was throwing if the key was null. However, the error message was very generic and didn't provide any useful information to the end user.

We are the only people who will be calling this method so if we are passing nulls/empty it means there is something wrong with our calling code. An assertion is enough to check that.